tensorflow-framework/src/test/java/org/tensorflow/framework/constraints/NonNegTest.java

```diff
@@ -0,0 +1,110 @@
+package org.tensorflow.framework.constraints;
+
+import org.junit.jupiter.api.Test;
+import org.tensorflow.Operand;
+import org.tensorflow.framework.utils.ND;
+import org.tensorflow.framework.utils.TestSession;
+import org.tensorflow.ndarray.DoubleNdArray;
+import org.tensorflow.ndarray.FloatNdArray;
+import org.tensorflow.ndarray.NdArrays;
+import org.tensorflow.ndarray.Shape;
+import org.tensorflow.op.Ops;
+import org.tensorflow.types.TFloat32;
+import org.tensorflow.types.TFloat64;
+
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+
+class NonNegTest {
+  private final TestSession.Mode[] tfModes = {TestSession.Mode.EAGER, TestSession.Mode.GRAPH};
+
+  private float[] getSampleArray() {
+    Random rand = new Random(3537L);
+    float[] result = new float[100 * 100];
+    for (int i = 0; i < result.length; i++) {
+      result[i] = rand.nextFloat() * 100 - 50;
+    }
+    result[0] = 0;
+    return result;
+  }
+
+  private double[] getSampleDArray() {
+    Random rand = new Random(3537L);
+    double[] result = new double[100 * 100];
+    for (int i = 0; i < result.length; i++) {
+      result[i] = rand.nextDouble() * 100 - 50;
+    }
+    result[0] = 0;
+    return result;
+  }
+
+  @Test
+  public void testTFloat32() {
+    float[] testValues = {0.1f, 0.5f, 3f, 8f, 1e-7f};
+    for (TestSession.Mode tfMode : tfModes)
+      try (TestSession session = TestSession.createTestSession(tfMode)) {
+        Ops tf = session.getTF();
+        final float[] array = getSampleArray();
+        Operand<TFloat32> weights = tf.reshape(tf.constant(array), tf.constant(Shape.of(100, 100)));
+        for (AtomicInteger i = new AtomicInteger();
+            i.get() < testValues.length;
+            i.getAndIncrement()) {
+          MinMaxNorm<TFloat32> instance =
```

Comment:
From this point forward, the test is erroneously a copy of `MinMaxNormTest`.

Reply:
My mistake, I'll fix.

Reply:
👍 done